File: chapter_01/python/sicpc1e01.py
## The solution of exercise 1.1
## Give the result printed by the interpreter in response to each express-
## ion. Assume that the sequence is to be evaluated in the order in which
## it is presented.

## Define some variables used later
a = 3
b = a + 1

result = [0] * 11
result[0] = 10
result[1] = 5 + 3 + 4
result[2] = 9 - 1
result[3] = 6 / 2
result[4] = ((2 * 4) + (4 - 6))
result[5] = (a + b + a * b)
result[6] = (a == b)
if b > a and b < a * b:
    result[7] = b
else:
    result[7] = a
    
if a == 4:
    result[8] = 6
elif b == 4:
    result[8] = 6 + 7 + a
else:
    result[8] = 25
    
if b > a:
    result[9] = b + 2
else:
    result[9] = a + 2
    
temp = 0
if a > b:
    temp = a
elif a < b:
    temp = b
else:
    temp = -1
result[10] = temp * (a + 1)

File: chapter_01/haskell/sicpc1e01.hs
-- The solution of exercise 1.1
-- Give the result printed by the interpreter in response to each express-
-- ion. Assume that the sequence is to be evaluated in the order in which
-- it is presented.

a = 3
b = a + 1

result = do { print 10;
              print (5 + 3 + 4);
              print (9 - 1);
              print (6 / 2);
              print ((2 * 4) + (4 - 6));
              print (a + b + a * b);
              print (a == b);
              print (if b > a && b < a * b
                     then b            
                     else a);
              print (if a == 4
                     then 6
                     else if b == 4
                          then 6 + 7 + a
                          else 25);
              print (if b > a
                     then b + 2
                     else a + 2);
              print (let temp = a + 1
                     in if a > b
                        then temp * a
                        else if a < b
                             then temp * b
                             else temp * (-1)); }


